internal/mesh/internal/controllers/xds/controller_test.go

```diff
@@ -1167,3 +1242,47 @@ func JSONToProxyTemplate(t *testing.T, json []byte) *pbmesh.ProxyStateTemplate {
 	require.NoError(t, err)
 	return proxyTemplate
 }
+
+func (suite *xdsControllerTestSuite) runTestCaseWithTenancies(testCase func(tenancy *pbresource.Tenancy)) {
+	for _, tenancy := range suite.tenancies {
+		suite.Run(suite.appendTenancyInfo(tenancy), func() {
+			defer suite.cleanup()
+			testCase(tenancy)
+		})
+	}
+}
+
+func (suite *xdsControllerTestSuite) appendTenancyInfo(tenancy *pbresource.Tenancy) string {
+	return fmt.Sprintf("%s_Namespace_%s_Partition", tenancy.Namespace, tenancy.Partition)
+}
+
+func (suite *xdsControllerTestSuite) cleanup() {
```

Comment:
If we use `t.Cleanup`, we can put all of these immediately after writing resources.

Reply:
I'll replace `defer` with `t.cleanup` then, thanks